### PR TITLE
Feature/mobile menu

### LINKED ISSUE #20: Mobile Menu
https://www.figma.com/file/DTJwC2mQ9kvyjM0171tqHR/airswap-web?node-id=13640%3A21079&t=qpsT15RjKpjhmpOy-0

Image

Should animate in right to left like airswap-web.

### PR DESCRIPTION
**Pull request by GiuntaLucas**
Mobile menu trigger by the menu/close button.

Fixes #20

image

### CODE REVIEW

**Review by piersss — CHANGES_REQUESTED** [review]
The correct way to start a PR:

https://github.com/airswap/airswap-marketplace/blob/main/CONTRIBUTING.md#pull-requests-prs

File: src/compositions/TopBar/MobileMenu/MobileMenu.scss [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    animation: fly-in-top ease-in-out 0.5s forwards;
+    font-weight: var(--fw-700);
+    
+    &--fly-out {
+        animation: fly-out-top ease-in-out 0.5s forwards;
+    }
+    div {
+        padding-top: 2.5rem;
+        padding-left: 1rem;
+    }
```
[the remark this answers is not in the document]
Comment (GiuntaLucas) [reply src/compositions/TopBar/MobileMenu/MobileMenu.scss:20]: I will replace the div by the NavLink and styles them

File: src/components/WalletInfo/WalletInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { FC } from 'react';
+
+import IconButton from '../../compositions/IconButton/IconButton';
+
+import './WalletInfo.scss';
+
+
+interface WalletInfoProps {
+  mode: 'BANNER' | 'MENU',
```
[the remark this answers is not in the document]
Comment (GiuntaLucas) [reply src/components/WalletInfo/WalletInfo.tsx:9]: This mode / boolean will be usefull to display / hide some informations / styles following the placement of this components.

I saw some screen from desktop with this kind of component.

File: src/components/Icon/icons/IconLaunch.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  <svg
+    className={className}
+    width="18"
+    height="18"
+    viewBox="0 0 18 18"
+    fill="none"
+    xmlns="http://www.w3.org/2000/svg"
+  >
+    <path
+      d="M16 16H2V2H9V0H2C1.46957 0 0.960859 0.210714 0.585786 0.585786C0.210714 0.960859 0 1.46957 0 2V16C0 16.5304 0.210714 17.0391 0.585786 17.4142C0.960859 17.7893 1.46957 18 2 18H16C17.1 18 18 17.1 18 16V9H16V16ZM11 0V2H14.59L4.76 11.83L6.17 13.24L16 3.41V7H18V0H11Z"
```
[the remark this answers is not in the document]
Comment (GiuntaLucas) [reply src/components/Icon/icons/IconLaunch.tsx:15]: What should i do ? It is the svg i received from the export of figma

File: src/components/WalletInfo/WalletInfo.scss [hunk 4]
```diff
@@ -0,0 +1,21 @@
+.wallet-info {
```
[the remark this answers is not in the document]
Comment (GiuntaLucas) [reply src/components/WalletInfo/WalletInfo.scss:1]: Yes i did, the .wallet-info class is the main class of the component. I use (wrongly) later some modificators.
Should i rename the main class ?

File: src/components/WalletInfo/WalletInfo.scss [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  &__menu {
+    background-color: var(--c-grey);
+    border: .06rem solid var(--c-dark-border);
+    border-radius: .3rem;
+  }
+  &__banner {
+    background-color: transparent;
+  }
+  padding: 1rem!important;
```
[the remark this answers is not in the document]
Comment (GiuntaLucas) [reply src/components/WalletInfo/WalletInfo.scss:10]: inside of the &__banner, i should use &--padding, and use this class to the right element, correct ?

File: src/compositions/TopBar/MobileMenu/MobileMenu.scss [hunk 6]
```diff
@@ -0,0 +1,39 @@
+.container {
+    width: 100vw;
+    height: calc(100vh - 3rem);
```
[the remark this answers is not in the document]
Comment (GiuntaLucas) [reply src/compositions/TopBar/MobileMenu/MobileMenu.scss:3]: I didn't no this unit 😮

File: src/components/Icon/icons/IconLaunch.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (piersss) [reply src/components/Icon/icons/IconLaunch.tsx:15]: Look at other svg's in icon list. Some are too long as well.

File: src/components/WalletInfo/WalletInfo.scss (hunk 4 above)
[the remark this answers is not in the document]
Comment (piersss) [reply src/components/WalletInfo/WalletInfo.scss:1]: .wallet-info is fine!

File: src/compositions/TopBar/MobileMenu/MobileMenu.scss [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    position: absolute;
+    left: 0;
+    top: 3rem;
+    display: flex;
+    flex-direction: column;
+    padding: 1.5rem;
+    background-color: var(--c-black);
+    animation: fly-in-top ease-in-out 0.5s forwards;
+    font-weight: var(--fw-700);
```
[the remark this answers is not in the document]
Comment (piersss) [reply src/compositions/TopBar/MobileMenu/MobileMenu.scss:12]: yup, `&__mobile-menu`

File: src/compositions/TopBar/MobileMenu/MobileMenu.tsx [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+      <WallelInfo mode="MENU" />
+      <div>My NFTs</div>
+      <div>My Activity</div>
+      <div>My Listed NFTs</div>
+      <div>List NFT</div>
+      <div>Project Discord</div>
+    </div>
+  ) : null;
+  return template;
+};
```
[the remark this answers is not in the document]
Comment (piersss) [reply src/compositions/TopBar/MobileMenu/MobileMenu.tsx:43]: its not necessary for a navigation menu

File: src/compositions/TopBar/MobileMenu/MobileMenu.scss (hunk 6 above)
[the remark this answers is not in the document]
Comment (piersss) [reply src/compositions/TopBar/MobileMenu/MobileMenu.scss:3]: its new!

File: src/components/WalletInfo/WalletInfo.scss (hunk 5 above)
[the remark this answers is not in the document]
Comment (piersss) [reply src/components/WalletInfo/WalletInfo.scss:10]: no idea what you mean, just dont use important

File: src/components/WalletInfo/WalletInfo.scss [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+  &--is-menu {
+    background-color: var(--c-grey);
+    border: .06rem solid var(--c-dark-border);
+    border-radius: .3rem;
+  }
+  &--is-banner {
+    background-color: transparent;
+  }
+  &--padding {
```
Comment (piersss) [inline src/components/WalletInfo/WalletInfo.scss:13]: https://github.com/airswap/airswap-marketplace/blob/main/CONTRIBUTING.md#naming-of-modifiers

File: src/components/WalletInfo/WalletInfo.scss [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+    padding: 1rem;
+  }
+   
+  &__img { 
+    width: 2.5rem;
```
Comment (piersss) [inline src/components/WalletInfo/WalletInfo.scss:18]:
```suggestion
    min-width: 2.5rem;
```

File: src/components/WalletInfo/WalletInfo.scss [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+    height: 2.5rem;
+    border: 1px solid gray; //temp css, replace later by a generate img from address
+  }
```
Comment (piersss) [inline src/components/WalletInfo/WalletInfo.scss:21]:
```suggestion
  }
  
```

File: src/components/WalletInfo/WalletInfo.scss [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+  &__address {
+    margin-left: 0.875rem;
+    flex-grow: 1;
+    font-weight: var(--fw-700);
+  }
```
Comment (piersss) [inline src/components/WalletInfo/WalletInfo.scss:26]:
```suggestion
  }
  
```

File: src/components/WalletInfo/WalletInfo.tsx [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+import './WalletInfo.scss';
+
+
+interface WalletInfoProps {
+  isBanner?: boolean
+  className?: string
+}
+
+const WallelInfo: FC<WalletInfoProps> = ({ isBanner, className = '' }) => {
+  const walletInfoClassName = classNames('wallet-info wallet-info--padding', {
```
Comment (piersss) [inline src/components/WalletInfo/WalletInfo.tsx:16]:
```suggestion
  const walletInfoClassName = classNames('wallet-info', {
```

Unnecessary modifier.

File: src/components/WalletInfo/WalletInfo.tsx [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+    'wallet-info--is-banner': isBanner,
+    'wallet-info--is-menu': !isBanner,
+  }, className);
+
+  return (
+    <div className={walletInfoClassName}>
+      <div className="wallet-info__img" />
+      <span className="wallet-info__address">swapthebestnfts.eth</span>
+      { isBanner ? (<IconButton icon="launch" text="" iconClassName="wallet-info__icon" />) : null }
```
Comment (piersss) [inline src/components/WalletInfo/WalletInfo.tsx:25]:
```suggestion
      {isBanner && (
        <IconButton
          hideLabel
          icon="launch"
          text="wallet"
          iconClassName="wallet-info__icon"
        />
      )}
```

File: src/compositions/TopBar/MobileMenu/MobileMenu.scss [hunk 15]
```diff
@@ -0,0 +1,16 @@
+.container {
```
Comment (piersss) [inline src/compositions/TopBar/MobileMenu/MobileMenu.scss:1]: .wallet-info {

File: src/compositions/TopBar/MobileMenu/MobileMenu.scss [hunk 16]
```diff
@@ -0,0 +1,16 @@
...
+    display: flex;
+    flex-direction: column;
+    padding: 1.5rem;
+    font-weight: var(--fw-700);
+    background-color: var(--c-black);
+    transform: translateY(0vh);
+    transition: transform 0.5s;
```
Comment (piersss) [inline src/compositions/TopBar/MobileMenu/MobileMenu.scss:8]:
```suggestion
    transition: transform 0.5s;
    overflow-y: auto;
    
```

File: src/compositions/TopBar/MobileMenu/MobileMenu.scss [hunk 17]
```diff
@@ -0,0 +1,16 @@
...
+    &--is-hidden {
+        transform: translateY(-100vh);
+    }
+    &__nav-link {
+        padding-top: 2.5rem;
```
Comment (piersss) [inline src/compositions/TopBar/MobileMenu/MobileMenu.scss:13]: Screenshot 2023-01-08 at 22 18 10

File: src/compositions/TopBar/MobileMenu/MobileMenu.tsx [hunk 18]
```diff
@@ -0,0 +1,33 @@
...
+
+import './MobileMenu.scss';
+
+interface MobileMenuProp {
+  isHidden: boolean;
+  className?: string;
+}
+
+const MobileMenu: FC<MobileMenuProp> = ({ isHidden, className = '' }) => {
+  const containerClassName = classNames('container', {
```
Comment (piersss) [inline src/compositions/TopBar/MobileMenu/MobileMenu.tsx:17]:
```suggestion
  const containerClassName = classNames('wallet-info', {
```

File: src/components/WalletInfo/WalletInfo.tsx [hunk 19]
```diff
@@ -0,0 +1,31 @@
...
+
+import './WalletInfo.scss';
+
+
+interface WalletInfoProps {
+  isBanner?: boolean
+  className?: string
+}
+
+const WallelInfo: FC<WalletInfoProps> = ({ isBanner, className = '' }) => {
```
Comment (piersss) [inline src/components/WalletInfo/WalletInfo.tsx:15]:
```suggestion
const WalletInfo: FC<WalletInfoProps> = ({ isBanner = false, className = '' }) => {
```

File: src/compositions/TopBar/MobileMenu/MobileMenu.tsx [hunk 20]
```diff
@@ -0,0 +1,33 @@
...
+  }, className);
+
+  return (
+    <div className={containerClassName}>
+      <WallelInfo isBanner={false} />
+      <NavLink className="container__nav-link" to="">My NFTs</NavLink>
+      <NavLink className="container__nav-link" to="">My Activity</NavLink>
+      <NavLink className="container__nav-link" to="">My Listed NFTs</NavLink>
+      <NavLink className="container__nav-link" to={`/${AppRoutes.listNft}`}>List NFT</NavLink>
+      <NavLink className="container__nav-link" to="">Project Discord</NavLink>
```
Comment (piersss) [inline src/compositions/TopBar/MobileMenu/MobileMenu.tsx:28]: This links to https://discord.com/invite/ecQbV7H so should be an `a`.

File: src/compositions/TopBar/TopBar.tsx [hunk 21]
```diff
@@ -1,37 +1,59 @@
...
+const TopBar: FC<TopBarProps> = ({ className = '' }) => {
+  const [mobileMenuIsVisible, setMobileMenuIsVisible] = useState<boolean>(false);
+
+  const topBarClassName = classNames('top-bar', {
+    'top-bar__mobile': mobileMenuIsVisible,
+  }, className);
+
+  const closeIconClassName = classNames('', {
+    'top-bar__close-icon': mobileMenuIsVisible,
+  });
```
Comment (piersss) [inline src/compositions/TopBar/TopBar.tsx:25]:
```suggestion
  const containerClassName = classNames('top-bar', {
    'top-bar--mobile-menu-is-visible': mobileMenuIsVisible,
  }, className);
```

File: src/compositions/TopBar/MobileMenu/MobileMenu.scss [hunk 22]
```diff
@@ -0,0 +1,16 @@
...
+    display: flex;
+    flex-direction: column;
+    padding: 1.5rem;
+    font-weight: var(--fw-700);
+    background-color: var(--c-black);
+    transform: translateY(0vh);
+    transition: transform 0.5s;
+    &--is-hidden {
+        transform: translateY(-100vh);
+    }
```
Comment (piersss) [inline src/compositions/TopBar/MobileMenu/MobileMenu.scss:11]:
```suggestion
    }

```

File: src/components/WalletInfo/WalletInfo.scss [hunk 23]
```diff
@@ -0,0 +1,31 @@
+.wallet-info {
+  display: flex;
+  align-items: center;
+
+  &--is-menu {
+    background-color: var(--c-grey);
+    border: .06rem solid var(--c-dark-border);
```
Comment (piersss) [inline src/components/WalletInfo/WalletInfo.scss:7]:
```suggestion
    border: 1px solid var(--c-dark-border);
```

File: src/components/WalletInfo/WalletInfo.scss [hunk 24]
```diff
@@ -0,0 +1,31 @@
...
+    border-radius: .3rem;
```
Comment (piersss) [inline src/components/WalletInfo/WalletInfo.scss:8]:
```suggestion
    border-radius: .3125rem;
```

File: src/compositions/TopBar/TopBar.scss [hunk 25]
```diff
@@ -10,7 +10,17 @@ $component: top-bar;
...
+    position: relative;
+    z-index: 1;
 
+    &__mobile-menu {
+        position: fixed;
+        left: 0;
+        top: 3rem;
+        width: 100vw;
+        height: calc(100svh - 3rem);
+        z-index: -1;
```
Comment (piersss) [inline src/compositions/TopBar/TopBar.scss:22]: I just added this to development, so you'll have to rebase to get it.

```suggestion
        z-index: -1;
        
        @include for-size(tablet-portrait-up) {
            display: none;
        }
```

File: src/compositions/TopBar/MobileMenu/MobileMenu.scss (hunk 16 above)
Comment (piersss) [inline src/compositions/TopBar/MobileMenu/MobileMenu.scss:8]:
```suggestion
    transition: transform 0.5s ease-out;
```

**Review by piersss — CHANGES_REQUESTED** [review]

File: src/compositions/TopBar/TopBar.scss (hunk 25 above)
Comment (GiuntaLucas) [reply src/compositions/TopBar/TopBar.scss:26]: This code is not yet on the main branch, still on your pr.

**Review by piersss — APPROVED** [review]
LGTM

**Pull request merged**